src/Adapter/MSTest.TestAdapter/Discovery/TypeEnumerator.cs

```diff
@@ -174,29 +190,29 @@ internal UnitTestElement GetTestFromMethod(MethodInfo method, bool isDeclaredInT
 
         testElement.Traits = traits.ToArray();
 
-        if (_reflectHelper.GetCustomAttribute<CssIterationAttribute>(method) is CssIterationAttribute cssIteration)
```

Comment:
GetCustomAttribute checks for a derived attribute (e.g. MyTestClass when we look for TestClass), and inherits (inspects parent types).

All our attributes allow inheritance, even though we don't always check for it.

Almost none of our attributes allow multiple, just testCategory, testProperty, and workItem. 

So getting first without checking that there is only Single is okay, because there never won't be more, unless you are using the attributes above.